crates/sui-indexer/src/models_v2/epoch.rs

```diff
@@ -33,6 +33,39 @@ pub struct StoredEpochInfo {
     pub epoch_commitments: Option<Vec<u8>>,
 }
 
+#[derive(Queryable)]
+pub struct QueryableEpochInfo {
+    pub epoch: i64,
+    pub first_checkpoint_id: i64,
+    pub epoch_start_timestamp: i64,
+    pub reference_gas_price: i64,
+    pub protocol_version: i64,
+    pub total_stake: i64,
+    pub storage_fund_balance: i64,
+    pub epoch_total_transactions: Option<i64>,
+    pub last_checkpoint_id: Option<i64>,
+    pub epoch_end_timestamp: Option<i64>,
+    pub storage_fund_reinvestment: Option<i64>,
+    pub storage_charge: Option<i64>,
+    pub storage_rebate: Option<i64>,
+    pub stake_subsidy_amount: Option<i64>,
+    pub total_gas_fees: Option<i64>,
+    pub total_stake_rewards_distributed: Option<i64>,
+    pub leftover_storage_fund_inflow: Option<i64>,
+    pub epoch_commitments: Option<Vec<u8>>,
+}
+
+#[derive(Queryable)]
+pub struct QueryableEpochSystemState {
+    pub epoch: i64,
+    pub system_state: Vec<u8>,
+}
+
+#[derive(Queryable)]
+pub struct QueryableEpochId {
```

Comment:
Is this needed?

Reply:
Ashok mentioned that sometimes only epoch ID is needed but more data had to be loaded, so I added this one, not exactly for the split per se.

Reply:
I think we can do without this one -- epochs are keyed by their ID, so if we were clever enough to spot that that was the only field being requested, then we could serve it without going to the DB at all.